pyface/font.py

```diff
@@ -0,0 +1,478 @@
+    This selects either 'oblique' or 'italic' variants typefaces of the given
+    font family.  If neither is wanted, the value is 'normal'.
+
+Size
+    The overall size of the glyphs. This can be expressed either as the
+    numeric size in points, or as a string such as "small" or "large".
+
+Variants
+    A set of additional font style specifiers, such as "small-caps",
+    "strikethrough", "underline" or "overline", where supported by the
+    underlying toolkit.
+
+Text Description
+----------------
+
+Frequently it is useful to specify a font by a descriptive string (or supply
+such a description to the user).  For these we follow the conventions in CSS
+where fonts are specfied by a string which specifies the weight, stretch,
+style and variants by text synonyms (in any order), followed by size in points
+and font family preferences (quoted if not a single word) and separated by
+commas.  Where the value is "normal" it can be omitted from the description.
+
+For example::
+
+    'italic bold 14pt Helvetica, Arial, sans-serif'
+    '36pt "Comic Sans"'
+
+When converting numeric values to string synonyms for display, the nearest
+value will be chosen.  This may mean that text descriptions may not be not
+idempotent when run through font selection multiple times.
+
+Font Specificiation Class
+-------------------------
+
+The Pyface Font class is a HasStrictTraits class which specifies a requested
+font.  It has traits for all of the font properties, plus additional utility
+methods that produce modifed versions of the font.
+
+It also has methods that convert the Font class to and from a toolkit Font
+class.
+
+"""
+import re
+
+from traits.api import (
+    BaseCFloat, DefaultValue, Enum, HasStrictTraits, List, Map, Set, Str,
+    TraitError, TraitType
+)
+from traits.trait_type import NoDefaultSpecified
+
+weights = {str(i): i for i in range(100, 1001, 100)}
+weights.update({
+    'thin': 100,
+    'extra-light': 200,
+    'ultra-ight': 200,
+    'light': 300,
+    'normal': 400,
+    'regular': 400,
+    'book': 400,
+    'medium': 500,
+    'semibold': 600,
+    'demibold': 600,
+    'demi': 600,
+    'bold': 700,
+    'heavy': 800,
+    'extra-bold': 800,
+    'ultra-bold': 800,
+    'black': 900,
+    'heavy': 900,
+    'extra-heavy': 1000,
+})
+
+stretches = {
+    'ultra-condensed': 100,
+    'ultracondensed': 100,
+    'extra-condensed': 200,
+    'extracondensed': 200,
+    'condensed': 300,
+    'semi-condensed': 400,
+    'semicondensed': 400,
+    'normal': 500,
+    'semi-expanded': 600,
+    'semiexpanded': 600,
+    'expanded': 700,
+    'extra-expanded': 800,
+    'extraexpanded': 800,
+    'ultra-expanded': 900,
+    'ultraexpanded': 900,
+}
+
+sizes = {
+    'xx-small': 7.0,
+    'x-small': 9.0,
+    'small': 10.0,
+    #'medium': 12.0,
+    'large': 14.0,
+    'x-large': 18.0,
+    'xx-large': 20.0,
+}
+
+styles = ['normal', 'italic', 'oblique']
+
+variants = ['small-caps', 'underline', 'strikethrough', 'overline']
```

Comment:
I'm personally in two minds about all caps for container module "constants".  If they are intended to be mutated (eg. a registry) then I don't like all-caps.  In this case it's probably fine (in fact, I should possibly make them tuples rather than lists.